project/plugins/sql_pull_data.py

```diff
@@ -43,10 +45,11 @@ def __call__(self, query: str):
             SQL Query:"""
         prompt = ChatPromptTemplate.from_template(template)
 
-        db = SQLDatabase.from_uri(self.config.get("sqlite_db_path"))
+        if self.db is None:
```

Comment:
This is to avoid recreating a new SQLDB instance?

Reply:
yes